clang/test/CXX/dcl.dcl/dcl.attr/dcl.attr.noreturn/p1.cpp

```diff
@@ -49,3 +49,15 @@ void check() {
   test_type(g);
   test_type(h); // expected-note {{instantiation}}
 }
+
+namespace GH63009 {
+struct S {
+  [[noreturn]] S() { throw int {}; }
+};
+
+int test_no_return_constructor() { S(); } // ok
```

Comment:
Can you also add a test case (in a different test file under SemaCXX) for:
```
struct S {
  [[noreturn]] S();
};

int foo();
int func() {
  S s;
  foo(); // Unreachable warning
} // No "does not return a value" warning
```

Reply:
Also, since we're here, can you add a test to ensure that destructors also work as expected? That can be done in a separate PR if you'd like.

Reply:
> Can you also add a test case (in a different test file under SemaCXX) for:
>
>```cpp
>struct S {
>  [[noreturn]] S();
>};
>
>int foo();
>int func() {
>  S s;
>  foo(); // Unreachable warning
>} // No "does not return a value" warning
>
>```

@AaronBallman thanks for the feedback. I briefly checked the issue yesterday, and the warning for this case appears at the end of the function. I'll take a closer look, maybe need to make some additional changes.

Reply:
@AaronBallman I've added tests.